hpx/runtime/applier/apply_helper.hpp

```diff
@@ -134,7 +135,10 @@ namespace hpx { namespace applier { namespace detail
             naming::address::address_type lva,
             threads::thread_priority priority, Ts &&... vs)
         {
-            if (this_thread::has_sufficient_stack_space() || hpx::is_pre_startup())
+            // Direct actions should be able to be executed from a non-HPX thread
+            // as well
+            if (this_thread::has_sufficient_stack_space() ||
+                hpx::threads::get_self_ptr() == nullptr)
```

Comment:
What we have now is consistent, I guess. It is just the special case of when we try to run a direct action inside of one of the parcel pool threads where I want it to be run directly as well (as in not on a HPX thread, but still execute it directly if it is a direct action).